### PR TITLE
fix: removing PodSecurityPolicy files from manifests folders

### LINKED ISSUE #633: pod-security-policy.yaml is being parsed as a pod
**Is this a request for help?**:
No
---

**Is this an ISSUE or FEATURE REQUEST?** (choose one):
ISSUE
---

**What version of aks-engine?**:
0.31.1
---

**Kubernetes version**:
1.13.3

**What happened**:
I created a cluster with rbac and pod security policies enabled then inspected the kubelet journal logs.

kubelet logs the following repeatedly:

```
Feb 28 22:05:20 master-27738982-0 kubelet[18862]: E0228 22:05:20.288964   18862 file.go:187] Can't process manifest file "/etc/kubernetes/manifests/pod-security-policy.yaml": /etc/kubernetes/manifests/pod-security-policy.yaml: couldn't parse as pod(invalid pod: &policy.PodSecurityPolicy{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"privileged", GenerateName:"", Namespace:"", SelfLink:"", UID:"", ResourceVersion:"", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string(nil), Annotations:map[string]string{"seccomp.security.alpha.kubernetes.io/allowedProfileNames":"*"}, OwnerReferences:[]v1.OwnerReference(nil), Initializers:(*v1.Initializers)(nil), Finalizers:[]string(nil), ClusterName:""}, Spec:policy.PodSecurityPolicySpec{Privileged:true, DefaultAddCapabilities:[]core.Capability(nil), RequiredDropCapabilities:[]core.Capability(nil), AllowedCapabilities:[]core.Capability{"*"}, Volumes:[]policy.FSType{"*"}, HostNetwork:true, HostPorts:[]policy.HostPortRange{policy.HostPortRange{Min:0, Max:65535}}, HostPID:true, HostIPC:true, SELinux:policy.SELinuxStrategyOptions{Rule:"RunAsAny", SELinuxOptions:(*core.SELinuxOptions)(nil)}, RunAsUser:policy.RunAsUserStrategyOptions{Rule:"RunAsAny", Ranges:[]policy.IDRange(nil)}, RunAsGroup:(*policy.RunAsGroupStrategyOptions)(nil), SupplementalGroups:policy.SupplementalGroupsStrategyOptions{Rule:"RunAsAny", Ranges:[]policy.IDRange(nil)}, FSGroup:policy.FSGroupStrategyOptions{Rule:"RunAsAny", Ranges:[]policy.IDRange(nil)}, ReadOnlyRootFilesystem:false, DefaultAllowPrivilegeEscalation:(*bool)(nil), AllowPrivilegeEscalation:true, AllowedHostPaths:[]policy.AllowedHostPath(nil), AllowedFlexVolumes:[]policy.AllowedFlexVolume(nil), AllowedUnsafeSysctls:[]string(nil), ForbiddenSysctls:[]string(nil), AllowedProcMountTypes:[]core.ProcMountType(nil)}}), please check config file.
```

**What you expected to happen**:
pod security policies provided by aks-engine are parsed and processed without errors

**How to reproduce it** (as minimally and precisely as possible):
Create a cluster with pod security policies and rbac enabled.

**Anything else we need to know**:
This is logged every 20 seconds.

### PR DESCRIPTION
**Pull request by allantargino**
**Reason for Change**:

Issue #633 describes a bug with kubelet trying to parse PodSecurityPolicy (PSP) file into a pod. The reason is because the PSP file were in a folder used by the kubelet to setup static pod definitions, causing the kubelet to try deploying it every 20s (default value for polling the definitions folder). 
The current cloudinit placeholder used to inject the PSP file is `MASTER_MANIFESTS_CONFIG_PLACEHOLDER`. There are 2 other places in the current templates we can inject it:
* MASTER_ADDONS_CONFIG_PLACEHOLDER
* MASTER_CONTAINER_ADDONS_PLACEHOLDER
Since `addons` seems more generic than `container addons` placeholder, I opted to move it to this location. 

**Issue Fixed**:

fixes #633 

**Requirements**:

- [X] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: parts/k8s/cloud-init/artifacts/cse_config.sh [hunk 1]
```diff
@@ -302,7 +302,7 @@ ensureJournal(){
 }
 
 ensurePodSecurityPolicy() {
```
Comment (jackfrancis) [inline parts/k8s/cloud-init/artifacts/cse_config.sh]: I think we can get rid of this `ensurePodSecurityPolicy` func (and anywhere that it is called) altogether now that we're putting it in the `addons/` directory. Manifests under `/etc/kubernetes/addons/` should be loaded automatically by the addon-manager component, so we just need to do all our "is psp enabled?" conditional logic during template generation, but not at provisioning runtime.

**Comment by jackfrancis** [comment]
Thanks for this @allantargino!

File: parts/k8s/cloud-init/artifacts/cse_config.sh (hunk 1 above)
Comment (allantargino) [reply parts/k8s/cloud-init/artifacts/cse_config.sh]: Great insight. It took me a while to figure it how to proper use the addon-manager (my bad!) but I really liked this approach. Thanks @jackfrancis and @CecileRobertMichon!

**Pull request merged**